src/inventory/chest_inventory.rs

```diff
@@ -141,72 +82,95 @@ impl Inventory for ChestInventory {
-            .alignment(ui::VAttach::Top, ui::HAttach::Center)
-            .size(icon_scale * 176.0, icon_scale * 96.0)
-            .texture("minecraft:gui/container/generic_54")
-            .create(ui_container);
-        basic_elements.push(player_inv_img);
-        let top_inv_img = ui::ImageBuilder::new()
-            .texture_coords((
-                0.0 / 256.0,
-                0.0 / 256.0,
-                176.0 / 256.0,
-                (rows * 18 + 17) as f64 / 256.0,
-            ))
-            .position(0.0, icon_scale * y)
-            .alignment(ui::VAttach::Top, ui::HAttach::Center)
-            .size(icon_scale * 176.0, icon_scale * (rows * 18 + 17) as f64)
-            .texture("minecraft:gui/container/generic_54")
-            .create(ui_container);
-        basic_elements.push(top_inv_img);
-        let scale = icon_scale / 2.0;
+        inventory_window.elements.push(vec![]); // Window texture
+        inventory_window.elements.push(vec![]); // Chest slots
+        inventory_window.elements.push(vec![]); // Base slots
         inventory_window.text_elements.push(vec![]);
+
+        let basic_elements = inventory_window.elements.get_mut(0).unwrap();
         let basic_text_elements = inventory_window.text_elements.get_mut(0).unwrap();
-        let title_text = ui::TextBuilder::new()
-            .alignment(VAttach::Top, HAttach::Center)
-            .scale_x(scale)
-            .scale_y(scale)
-            .position(
-                icon_scale
-                    * -(176.0 / 2.0 - 8.0 - renderer.ui.lock().size_of_string(&self.name) / 4.0),
-                icon_scale * (6.0 + y),
-            )
-            .text(&self.name)
-            .colour((64, 64, 64, 255))
-            .shadow(false)
-            .create(ui_container);
-        basic_text_elements.push(title_text);
-        let inventory_text = ui::TextBuilder::new()
-            .alignment(VAttach::Top, HAttach::Center)
-            .scale_x(scale)
-            .scale_y(scale)
-            .position(
-                icon_scale
-                    * -(176.0 / 2.0 - 8.0 - renderer.ui.lock().size_of_string("Inventory") / 4.0),
-                icon_scale * (6.0 + y + (rows * 18 + 13) as f64),
-            )
-            .text("Inventory")
-            .colour((64, 64, 64, 255))
-            .shadow(false)
-            .create(ui_container);
-        basic_text_elements.push(inventory_text);
-        inventory_window.elements.push(vec![]);
-        self.update_icons(renderer);
-        self.hud_context
-            .clone()
-            .read()
-            .dirty_slots
-            .store(true, Ordering::Relaxed);
+
+        let icon_scale = Hud::icon_scale(renderer.clone()) as i32;
+        let chest_grid_height = icon_scale * (self.rows as i32 * 18 + 17);
+        let inventory_grid_height = icon_scale * 96;
+        let total_height = chest_grid_height + inventory_grid_height;
+        let total_width = icon_scale * 176;
```

Comment:
you are using `176` as a constant in multiple places here, maybe declaring a constant could make what you want to do a bit more clear, same for `18`, `17` and `96`